File: app/src/main/java/io/github/kutaiba_nezar_kashmar/multiverse_of_geeks/domain/response/TrendingResponse.java
package io.github.kutaiba_nezar_kashmar.multiverse_of_geeks.domain.response;

import java.util.ArrayList;

import io.github.kutaiba_nezar_kashmar.multiverse_of_geeks.domain.Trending;

public class TrendingResponse
{
  private int id;
  private int vote_count;
  private String original_language;
  private String poster_path;
  private String title;
  private float vote_average;
  private String overview;
  private String release_date;
  private float popularity;
  private String media_type;
  private ArrayList<Trending> results = null;

  public Trending getTrending()
  {
    return new Trending(id, vote_count, original_language, poster_path, title,
        vote_average, overview, release_date, popularity, media_type);
  }

  public ArrayList<Trending> getResults()
  {
    return results;
  }
}

File: app/src/main/java/io/github/kutaiba_nezar_kashmar/multiverse_of_geeks/model/domain/response/TrendingResponse.kt
package io.github.kutaiba_nezar_kashmar.multiverse_of_geeks.model.domain.response

import io.github.kutaiba_nezar_kashmar.multiverse_of_geeks.model.domain.Trending
import java.util.ArrayList

class TrendingResponse
{
    val id: Int = 0
    val vote_count: Int = 0
    val original_language: String = ""
    val poster_path: String = ""
    val title: String = ""
    val vote_average: Float = 0.0F
    val overview: String = ""
    val release_date: String = ""
    val popularity: Float = 0.0F
    val media_type: String = ""
    val results: List<Trending> = ArrayList()

    fun getTrending(): Trending
    {
        return Trending(
            id,
            vote_count,
            original_language,
            poster_path,
            title,
            vote_average,
            overview,
            release_date,
            popularity,
            media_type
        )
    }
}